### PR TITLE
feat: Add support for the other nim language server

### PR DESCRIPTION
**Pull request by marcusramberg**
---
name: Pull Request
about: Submit a pull request
title: ''
---

### CODE REVIEW

File: lua/lspconfig/server_configurations/nim-langserver.lua [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    single_file_support = true,
+  },
+  docs = {
+    description = [[
+https://github.com/nim-lang/langserver 
+
+
+`nim-langserver` can be installed via the `nimble` package manager:
+```sh
+```
```
Comment (glepnir) [inline lua/lspconfig/server_configurations/nim-langserver.lua:19]:
```suggestion
nimble install nimlangserver
```

**Review by glepnir — CHANGES_REQUESTED** [review]
change file name `nim-langserver` to `nim_ls` or `nim_langserver`

File: lua/lspconfig/server_configurations/nim-langserver.lua (hunk 1 above)
Comment (marcusramberg) [reply lua/lspconfig/server_configurations/nim-langserver.lua]: > change file name `nim-langserver` to `nim_ls` or `nim_langserver`

Given therese a nimls.lua already, I renamed to nim_langserver

File: lua/lspconfig/server_configurations/nim_langserver.lua [hunk 2]
```diff
@@ -0,0 +1,23 @@
+local util = require 'lspconfig.util'
+
```
Comment (glepnir) [inline lua/lspconfig/server_configurations/nim_langserver.lua:2]: add windows support here .

Comment (marcusramberg) [reply lua/lspconfig/server_configurations/nim_langserver.lua:2]: Thank you for the review. Could you give me a pointer about what is required for windows support. I based this on nimls.lua, so I guess the same change would be needed in that module for windows support as well?

Comment (glepnir) [inline lua/lspconfig/server_configurations/nim_langserver.lua:2]:
```suggestion
local bin_name = 'nimlangserver'

local cmd = { bin_name }

if vim.fn.has 'win32' == 1 then
  cmd = { 'cmd.exe', '/C', bin_name }
end
```

File: lua/lspconfig/server_configurations/nim_langserver.lua [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+return {
+  default_config = {
+    cmd = { 'nimlangserver' },
```
Comment (glepnir) [inline lua/lspconfig/server_configurations/nim_langserver.lua:5]:
```suggestion
    cmd = cmd,
```

**Comment by marcusramberg** [comment]
Thanks! Rolled that change into my PR and added it for nimls as well.

File: lua/lspconfig/server_configurations/nim_langserver.lua [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+local bin_name =   'nimlangserver'
+local cmd = { bin_name }
+
+if vim.fn.has 'win32' == 1 then
+  cmd = { 'cmd.exe', '/C', bin_name }
+end
+
+return {
+  default_config = {
+    cmd = { cmd },
```
Comment (glepnir) [inline lua/lspconfig/server_configurations/nim_langserver.lua:11]:
```suggestion
    cmd = cmd,
```

**Pull request merged**